### PR TITLE
chore: added deep/universal links and in-app in kotlin sample

### PR DESCRIPTION
**Pull request by Shahroz16**
Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-android/develop/docs/dev-notes/GIT-WORKFLOW.md).
- [ ] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request.
- [ ] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1).
- [ ] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After pull request is approved, and you determine it's ready **add the label "Ready to merge"** to the pull request and it will automatically be merged.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -81,3 +81,4 @@ lint/tmp/
 
 # Android Profiling
 *.hprof
+samples/kotlin_compose/src/google-services.json
```
Comment (Shahroz16) [inline .gitignore:84]: due to nested `gitignore` this file was being tracked no matter what i did. So ignroing it in root gitignore

File: samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/data/sdk/InAppMessageEventListener.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        arguments: Map<String, String>?
+    ) {
+        CustomerIO.instance().track(
+            "in-app message action",
+            HashMap<String, String>().apply {
+                arguments?.let { putAll(it) }
+                put("event-name", eventName)
+                put("message-id", message.messageId)
+                val deliveryId = message.deliveryId
+                put("delivery-id", deliveryId ?: "NULL")
```
Comment (mrehan27) [inline samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/data/sdk/InAppMessageEventListener.kt:61]:
```suggestion
                put("delivery-id", message.deliveryId ?: "NULL")
```

File: samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/util/Logger.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        Log.d(TAG, message)
+    }
+
+    fun v(message: String) {
+        v(message, null)
+    }
+
+    fun v(message: String, throwable: Throwable?) {
+        Log.v(TAG, message, throwable)
+    }
```
Comment (mrehan27) [inline samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/util/Logger.kt:20]: We could use default parameters here 🤔 

```suggestion
    fun v(message: String, throwable: Throwable? = null) {
        Log.v(TAG, message, throwable)
    }
```

File: samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/util/Logger.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    fun e(message: String) {
+        e(message, null)
+    }
+
+    fun e(message: String, throwable: Throwable?) {
+        Log.e(TAG, message, throwable)
+    }
```
Comment (mrehan27) [inline samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/util/Logger.kt:28]: Same 🤓 

```suggestion
    fun e(message: String, throwable: Throwable? = null) {
        Log.e(TAG, message, throwable)
    }
```

**Review by mrehan27 — APPROVED** [review]
Minor tweaks needed, rest looks good

**Pull request merged**